components/proxy/src/test/java/com/hotels/styx/startup/StyxServerComponentsTest.java

```diff
@@ -0,0 +1,133 @@
+import com.hotels.styx.api.HttpInterceptor.Chain;
+import com.hotels.styx.api.configuration.Configuration;
+import com.hotels.styx.api.metrics.codahale.CodaHaleMetricRegistry;
+import com.hotels.styx.api.plugins.spi.Plugin;
+import com.hotels.styx.api.service.spi.StyxService;
+import com.hotels.styx.proxy.plugin.NamedPlugin;
+import com.hotels.styx.startup.StyxServerComponents.LoggingSetUp;
+import com.hotels.styx.support.api.BlockingObservables;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Map;
+
+import static com.hotels.styx.api.HttpRequest.Builder.get;
+import static com.hotels.styx.api.HttpResponse.Builder.response;
+import static com.hotels.styx.proxy.plugin.NamedPlugin.namedPlugin;
+import static com.hotels.styx.support.matchers.IsOptional.isValue;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.stream.Collectors.toList;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static rx.Observable.just;
+
+// TODO it is still difficult to decouple + test the configured plugins + services
+public class StyxServerComponentsTest {
+    @Test
+    public void setsUpLoggingOnBuild() {
+        LoggingSetUp loggingSetUp = mock(LoggingSetUp.class);
+
+        new StyxServerComponents.Builder()
+                .styxConfig(new StyxConfig())
+                .loggingSetUp(loggingSetUp)
+                .build();
+
+        verify(loggingSetUp).setUp(any(Environment.class));
+    }
+
+    @Test
+    public void loadsPlugins() {
+        Chain chain = mock(Chain.class);
+
+        List<NamedPlugin> pluginsForBuilder = ImmutableList.of(
+                namedPlugin("plugin1", stubPlugin("MyResponse1")),
+                namedPlugin("plugin2", stubPlugin("MyResponse2")));
+
+        StyxServerComponents components = new StyxServerComponents.Builder()
+                .styxConfig(new StyxConfig())
+                .plugins(env -> pluginsForBuilder)
+                .build();
+
+        List<NamedPlugin> plugins = ImmutableList.copyOf(components.plugins());
+        List<String> names = plugins.stream().map(NamedPlugin::name).collect(toList());
+
+        assertThat(names, contains("plugin1", "plugin2"));
+
+        List<String> responses = plugins.stream()
+                .map(plugin -> plugin.intercept(get("/").build(), chain))
+                .map(BlockingObservables::waitForResponse)
+                .map(resp -> resp.bodyAs(UTF_8))
+                .collect(toList());
+
+        assertThat(responses, contains("MyResponse1", "MyResponse2"));
+    }
+
+    @Test
+    public void loadsServices() {
```

Comment:
This test is not testing for service loading aspect. It is only verifying that additional services are added to the exposed services list. Consider renaming this method to "exposesAdditionalServices".